ztoc/toc_builder.go

```diff
@@ -150,8 +152,12 @@ func metadataFromTarReader(r io.Reader) ([]FileMetadata, compression.Offset, err
 			Xattrs:             hdr.PAXRecords,
 		}
 		md = append(md, metadataEntry)
+		tarHeaderOffset = metadataEntry.UncompressedOffset + metadataEntry.UncompressedSize
```

Comment:
Would the tar header offset not be the fixed 512 bytes before the uncompressed offset?

Reply:
This does set the tarHeaderOffset for the _next_ file. The actual tar header itself is 512 bytes, but that doesn't account for PAX records which are encoded as a special file. To make sure we don't get it wrong, we just pass all the bytes between files to the go tar header parser and make sure that all the bytes were consumed.